icu4c/source/tools/upropdump/upropdump.cpp

```diff
@@ -0,0 +1,178 @@
+
+int main(int argc, char* argv[]) {
+
+    U_MAIN_INIT_ARGS(argc, argv);
+
+    /* preset then read command line options */
+    options[OPT_DESTDIR].value=u_getDataDirectory();
+    argc=u_parseArgs(argc, argv, UPRV_LENGTHOF(options), options);
+
+    if(options[OPT_VERSION].doesOccur) {
+        printf("upropdump version %s, ICU tool to write Unicode property .toml files\n",
+               U_ICU_DATA_VERSION);
+        printf("%s\n", U_COPYRIGHT_STRING);
+        exit(0);
+    }
+
+    /* error handling, printing usage message */
+    if(argc<0) {
+        fprintf(stderr,
+            "error in command line argument \"%s\"\n",
+            argv[-argc]);
+    } else if(argc<2) {
+        argc=-1;
+    }
+
+    if(argc<0 || options[OPT_HELP_H].doesOccur || options[OPT_HELP_QUESTION_MARK].doesOccur) {
+        FILE *stdfile=argc<0 ? stderr : stdout;
+        fprintf(stdfile,
+            "usage: %s [-options] properties...\n"
+            "\tdump Unicode property data to .toml files\n"
+            "options:\n"
+            "\t-h or -? or --help  this usage text\n"
+            "\t-V or --version     show a version message\n"
+            "\t-c or --copyright   include a copyright notice\n"
+            "\t-d or --destdir     destination directory, followed by the path\n"
+            "\t-v or --verbose     Turn on verbose output\n"
+            "\t-q or --quiet       do not display warnings and progress\n",
+            argv[0]);
+        return argc<0 ? U_ILLEGAL_ARGUMENT_ERROR : U_ZERO_ERROR;
+    }
+
+    /* get the options values */
+    haveCopyright = options[OPT_COPYRIGHT].doesOccur;
+    const char *destdir = options[OPT_DESTDIR].value;
+    VERBOSE = options[OPT_VERBOSE].doesOccur;
+    QUIET = options[OPT_QUIET].doesOccur;
+
+    for (int i=1; i<argc; i++) {
+        auto propName = argv[i];
+        UProperty propEnum = u_getPropertyEnum(propName);
+        if (propEnum == UCHAR_INVALID_CODE) {
+            std::cerr << "Error: Invalid property alias: " << propName << std::endl;
+            exit(1);
+        }
+
+        ErrorCode status;
+        CharString outFileName;
+        if (destdir != nullptr || *destdir != 0) {
+            outFileName.append(destdir, status).ensureEndsWithFileSeparator(status);
+        }
+        outFileName.append(propName, status);
+        outFileName.append(".toml", status);
+        handleError(status, propName);
+
+        FILE* f = fopen(outFileName.data(), "w");
+        if (f == nullptr) {
+            std::cerr << "Unable to open file: " << outFileName.data() << std::endl;
+            exit(1);
+        }
+        if (!QUIET) {
+            std::cout << "Writing to: " << outFileName.data() << std::endl;
+        }
+
+        if (haveCopyright) {
+            usrc_writeCopyrightHeader(f, "#", 2021);
+        }
+        usrc_writeFileNameGeneratedBy(f, "#", propName, "upropdump.cpp");
+
+        if (propEnum < UCHAR_BINARY_LIMIT) {
+            dumpBinaryProperty(propEnum, f);
+        } else {
```

Comment:
You should check `UCHAR_INT_START <= propEnum && propEnum <= UCHAR_INT_LIMIT` and otherwise fail.

Reply:
93ca5257c52e152e4b8dee509d9e487f7b7dea58